digitaltwins/service/src/cl/digital_twin_service_client.ts

```diff
@@ -191,114 +138,28 @@ export class DigitalTwinServiceClient {
-  }
-
-  /**
-   * @method getModel                                 module: azure-iot-digitaltwins-service.DigitalTwinServiceClient.getModel
-   * @description                                     Retrieve a Digital Twin Model.
-   * @param {string} modelId                          The model Id of the requested model.
-   * @returns DigitalTwinGetDigitalTwinModelResponse  The return object containing the Model plus the HttpResponse.
+   * @method invokeComponentCommand                     module: azure-iot-digitaltwins-service.DigitalTwinServiceClient.invokeComponentCommand
+   * @description                                       Invoke a command on an component of a particular device and get the result of it.
+   * @param {string} digitalTwinId                      The digital twin Id of the given device.
+   * @param {string} componentName                      The component's name.
+   * @param {string} commandName                        The command's name.
+   * @param {string} argument                           The argument of a command.
+   * @returns DigitalTwinInvokeComponentCommandResponse The result of the invoked command containing the result, status code, request ID and the parsed HttpResponse.
    * @memberof DigitalTwinServiceClient
    */
-  getModel(modelId: string): Promise<DigitalTwinGetDigitalTwinModelResponse>;
-  getModel(modelId: string, callback: TripleValueCallback<Model, msRest.HttpOperationResponse>): void;
-  getModel(modelId: string, callback?: TripleValueCallback<Model, msRest.HttpOperationResponse>): void | Promise<DigitalTwinGetDigitalTwinModelResponse> {
-    /*Codes_SRS_NODE_DIGITAL_TWIN_SERVICE_CLIENT_12_008: [The `getModel` method shall call the `getDigitalTwinModel` method of the protocol layer with the given argument.]*/
-    /*Codes_SRS_NODE_DIGITAL_TWIN_SERVICE_CLIENT_12_009: [The `getModel` method shall call the callback with an error parameter if a callback is passed..]*/
-    /*Codes_SRS_NODE_DIGITAL_TWIN_SERVICE_CLIENT_12_010: [The `getModel` method shall return error if the method of the protocol layer failed.]*/
-    /*Codes_SRS_NODE_DIGITAL_TWIN_SERVICE_CLIENT_12_022: [The `getModel` method shall return a promise if there is no callback passed.]*/
-    return tripleValueCallbackToPromise<Model, msRest.HttpOperationResponse, DigitalTwinGetDigitalTwinModelResponse>((_callback) => {
-      this._pl.digitalTwin.getDigitalTwinModel(modelId, (err, result, request, response) => {
-        _callback(err as Error, result, response);
-      });
-    }, (result, response) => createResultWithHttpOperationResponse<Model, DigitalTwinGetDigitalTwinModelResponse>(result, response), callback as TripleValueCallback<Model, msRest.HttpOperationResponse>);
+  invokeComponentCommand(digitalTwinId: string, componentName: string, commandName: string, argument: string): Promise<DigitalTwinInvokeComponentCommandResponse>{
```

Comment:
>invokeComponentCommand [](start = 2, length = 22)

They didn't balk at two api for invoking commands?

Reply:
We have invokeComponentCommand and invokeCommand. Is that answers your question?

---
In reply to: [452337338](https://github.com/Azure/azure-iot-sdk-node/pull/844#discussion_r452337338) [](ancestors = 452337338)